### PR TITLE
Display plant labels.

### PR DESCRIPTION
**Pull request by badnames**
## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (Bushuo) [inline frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:27]: Please use the `ExtendedPlantsSummary` class from temmey here.
Maybe you can extract a common component from the resulting code here and in PlantingElement.
The tooltips should look the same IMO.
Would be nice if both code could be declarative.

This is not your fault, but he tooltip in PlantingElement should not generate all the text and layout on every mouse move.
The mouse move handler was initially only to reposition the tooltip.

Also, if the lables are active the tooltip can stay hidden.

**Review by Bushuo — CHANGES_REQUESTED** [review]
Hopefully just a minor change.
I can help, just say :)

**Comment by markus2330** [comment]
Great addition!

In order of importance:

- [ ] When selecting the plant, the label shouldn't be included. Fixing this would also fix a bug (Click on plant with label, disable label, then resize: the plant will not be round anymore)
- [ ] The name should be below the plants
- [ ] Can you make the distance dependent on the size of the plant? (So that it is always directly below the circle)

@Bushuo thx for the review! I fully agree that we shouldn't have duplicated rendering of the plant names. Depending on if we can easily have changes to the normal rendering of plant names (without duplicating any formatting code):

- [ ] if there is a common name, we actually don't need a latin name to be shown
- [ ] if this isn't easily possible (reusing the formatting from temmey), maybe it is possible to have a line break after the common name?

**Comment by markus2330** [comment]
Btw. the branch name is confusing and it seems like you didn't use the latest commit from master when branching off, at least #710 was present in this PR but is not on dev.permaplant.net?

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (badnames) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:27]: Great suggestions, thanks. I will make sure to ask if I need help :+1:.

**Comment by badnames** [comment]
> Btw. the branch name is confusing and it seems like you didn't use the latest commit from master when branching off, at least #710 was present in this PR but is not on dev.permaplant.net?

I created the branch yesterday by clicking 'create branch' in the GitHub issue so unless something was merged since yesterday this branch should theoretically be up to date.
The branch name is confusing, because I created it using the wrong issue without noticing...

I have linked the correct issue and changes the title of the PR to clear things up a bit.

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (temmey) [inline frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:22]: We discussed the presentation of 'plantname' in labels in issue #693. 
Do we want to keep this approach for simplicity's sake?
However, there are cases when a plant doesn't have a 'commonName', only a 'unique_name

Comment (markus2330) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:22]: The full plant name is described in doc/database/hierarchy.md and must be exactly like that (as also implemented by @temmey).

> Do we want to keep this approach for simplicity's sake?

Yes, it would be enough, see for options how to implement in https://github.com/ElektraInitiative/PermaplanT/pull/737#issuecomment-1656738605

Unfortunately, there was no response from @badnames yet.

> However, there are cases when a plant doesn't have a 'commonName', only a 'unique_name

@temmey Could you please document this case also in doc/database/hierarchy.md (as written in #693)

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
There is a formatting problem in this PR. Do you have pre-commit configured locally?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great addition!

In order of importance:

- [x] When selecting the plant, the label shouldn't be included. Fixing this would also fix a bug (Click on plant with label, disable label, then resize: the plant will not be round anymore)
- [x] The name should be below the plants
- [ ] Can you make the distance dependent on the size of the plant? (So that it is always directly below the circle)

@Bushuo thx for the review! I fully agree that we shouldn't have duplicated rendering of the plant names. Depending on if we can easily have changes to the normal rendering of plant names (without duplicating any formatting code):

- [ ] if there is a common name, we actually don't need a latin name to be shown
- [ ] if this isn't easily possible (reusing the formatting from temmey), maybe it is possible to have a line break after the common name?

**Comment by badnames** [comment]
image

@markus After some tinkering I managed to get the labels to your requested position.

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (badnames) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx]: The plant label is now generated using ExtendedPlantsSummary by @temmey.

**Comment by badnames** [comment]
jenkins build please

**Comment by badnames** [comment]
jenkins build please

**Comment by markus2330** [comment]
@badnames thank you, looks beautiful! Unfortunately, there is a merge conflict now.

**Comment by badnames** [comment]
> Maybe you can extract a common component from the resulting code here and in PlantingElement.
The tooltips should look the same IMO.
Would be nice if both code could be declarative.

> This is not your fault, but he tooltip in PlantingElement should not generate all the text and layout on every mouse move.
The mouse move handler was initially only to reposition the tooltip.

@Bushuo Could you put this request into its own issue?
I think it would be more appropriate to address this as a seperate PR.

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (Bushuo) [inline frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:21]: I don't think you need labelWidth here, but maybe I am missing something.

**Review by Bushuo — APPROVED** [review]
Looks good

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (Bushuo) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:21]: But you need the if because of the dependency.
I think both are unnecessary.

Comment (Bushuo) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:21]: To explain my reasoning, the effect is only run if `labelRef` or `labelWidth` changes.
`labelRef`  only changes on the initial render, but `labelWidth` changes every time `setLabelWidth` is called.
The `if` prevents the infinite rerender loop.
So if we exclude `labelWidth` from the dependencies and the `if` we are still left with the same functionality.

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
>   @Bushuo @markus2330 I think it would be more appropriate to address this as a seperate PR (mainly to keep this one short and simple).

The point is: we cannot merge this before the release if it somehow creates a regression.

New features etc. can always be put to a new PR but everything related to code quality (regressions, documentation, unit tests, ...) should be always fixed within PRs.

So in case of doubt, we better merge this after the release.

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (badnames) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:21]: Oh ok i thought the extra if check was necessary to avoid infinite loops due to having a setState call inside a useEffect hook.

File: doc/CHANGELOG.md
Comment (markus2330) [inline doc/CHANGELOG.md:19]: Something went wrong with the rebase. (This is also shown like this with `git diff` so it is not a bug of GitHub.)

**Comment by badnames** [comment]
> > @Bushuo @markus2330 I think it would be more appropriate to address this as a seperate PR (mainly to keep this one short and simple).
> 
> The point is: we cannot merge this before the release if it somehow creates a regression.
> 
> New features etc. can always be put to a new PR but everything related to code quality (regressions, documentation, unit tests, ...) should be always fixed within PRs.
> 
> So in case of doubt, we better merge this after the release.

Everything should now be implemented as requested.

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (markus2330) [inline frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:40]: If there is no common name, nothing is shown but the latin name should be shown, e.g. `Abies mariesii`. It looks like ExtendedPlantsSummary is not used?

File: frontend/src/features/map_planning/components/BaseStage.tsx
Comment (markus2330) [inline frontend/src/features/map_planning/components/BaseStage.tsx:241]: It looks to me that the mouse-over text is now bigger?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Looks much better! Still a few problems, though.

File: frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx
Comment (badnames) [reply frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx:40]: Done :+1:

File: frontend/src/features/map_planning/components/BaseStage.tsx
Comment (badnames) [reply frontend/src/features/map_planning/components/BaseStage.tsx:245]: Yes I had to make it larger because the tooltip would be kind of tiny otherwise.

**Comment by badnames** [comment]
jenkins build please

**Comment by markus2330** [comment]
jenkins build please

**Review by 4ydan — APPROVED** [review]
LGTM. One minor thing I realized is the label doesnt scale with the size of the icon. So when the plant is very small the label is proportionally big, but maybe thats not an issue.

**Review by markus2330 — APPROVED** [review]
Looks very nice now.

I wouldn't make the size of the font smaller, otherwise I could not read it.

There are wrong changes shown in the GitHub UI, maybe rebase helps.

**Pull request merged**
